data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessor.java

```diff
@@ -54,7 +54,7 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     }
 
                     try {
-                        final String key = entry.getKey();
+                        final String key = (entry.getKey() == null) ? null : recordEvent.formatString(entry.getKey());
```

Comment:
To support expressions in the format, you have to use this method (https://github.com/opensearch-project/data-prepper/blob/1aede50db3ba45e0e89876b035d236544335380c/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java#L325)

Reply:
Updated PR. 
Didn't realize we have this now. Thanks for the catch and help with debugging.